include/uart.h

```diff
@@ -64,6 +65,62 @@ extern "C" {
  */
 #define UART_ERROR_BREAK    (1 << 3)
 
+/**
+ * @brief UART controller configuration structure
+ *
+ * @param baudrate  Baudrate setting in bps
+ * @param parity    Parity bit, use @ref uart_config_parity
+ * @param stop_bits Stop bits, use @ref uart_config_stop_bits
+ * @param data_bits Data bits, use @ref uart_config_data_bits
+ * @param flow_ctrl Flow control setting, use @ref uart_config_flow_control
+ */
+struct uart_config {
+	u32_t baudrate;
+	u8_t parity;
+	u8_t stop_bits;
+	u8_t data_bits;
+	u8_t flow_ctrl;
+};
+
+/** @brief Parity modes */
+enum uart_config_parity {
+	UART_CFG_PARITY_NONE,
+	UART_CFG_PARITY_ODD,
+	UART_CFG_PARITY_EVEN,
+	UART_CFG_PARITY_MARK,
+	UART_CFG_PARITY_SPACE,
+};
+
+/** @brief Number of stop bits. */
+enum uart_config_stop_bits {
+	UART_CFG_STOP_BITS_0_5,
+	UART_CFG_STOP_BITS_1,
+	UART_CFG_STOP_BITS_1_5,
+	UART_CFG_STOP_BITS_2,
+};
+
+/** @brief Number of data bits. */
+enum uart_config_data_bits {
+	UART_CFG_DATA_BITS_5,
```

Comment:
Not sure if there is any value in doing:

`UART_CFG_DATA_BITS_5 = 5,`

Reply:
That would make it differ from parity and stop bits, I don't see how this approach could be used for stop bits for example